File: signature/swiper.go
package signature

var SwiperSignature = Signature{
	Name:        "Swiper",
	Description: "Most modern mobile touch slider and framework with hardware accelerated transitions.",
	Cpe:         "cpe:/a:swiperjs:swiper",

	DetectPattern: DetectPattern{
		URLs: []string{
			"npm/swiper@",
		},
		Bodies: []string{
			`\* Swiper `,
		},
	},
	VersionPattern: VersionPattern{
		URLs: []string{
			`npm/swiper@(\d+\.\d+\.\d+)/`,
		},
		Bodies: []string{
			`\* Swiper (\d+\.\d+\.\d+)`,
		},
	},
}

File: src/signatures/technologies/swiper.ts
import type { Signature } from "../_types.js";

export const swiperSignature: Signature = {
  name: "Swiper",
  description:
    "Most modern mobile touch slider and framework with hardware accelerated transitions",
  cpe: "cpe:/a:swiperjs:swiper",
  rule: {
    confidence: "high",
    urls: ["swiper[@.-](\\d+\\.\\d+\\.\\d+)?"],
    bodies: ["Swiper (\\d+\\.\\d+\\.\\d+)?"],
  },
};
